### ISSUE TITLE
Checkbox functionality

### ISSUE THREAD

**Issue by kenshin-takahiro** [BaldissaraMatheus/Tasks.md#76]
It would be great if you could enable checkboxes in the notes section of the tasks.
As you probably know the markdown syntax is this:
`- [ ] `

It would look like this:
- [ ] Subtask 1
- [ ] Subtask 2
- [X] Subtask 3

**Comment by BaldissaraMatheus** [comment]
Yeah, that's something I plan to work on very soon. I could be wrong, but it looks like the editor we're using does not support checkboxes. If that's the case we'll need to switch to a different editor

Also reported in #70

**Comment by BaldissaraMatheus** [comment]
It seems like StacksEditor, the editor we are currently using, implements the CommonMark specification, which does not support task list / checkboxes. So we'll have to switch to a different editor [that supports GFM](https://gist.github.com/vimtaai/99f8c89e7d3d02a362117284684baa0f#task-lists). This change will require some effort so I will not be able to deliver for the next release (which is going to be done this weekend), but is certainly going to be the focus for the next one

**Comment by kenshin-takahiro** [comment]
Looking forward to it for sure! Also if you can eventually show in the front of the card the amount of checkboxes that exist inside and how many of them are checked (i.e. 2/3, similar to how planka does it).

**Comment by BaldissaraMatheus** [comment]
I'd appreciate any suggestions on editors that works similarly as the one we currently use (must have markdown and rich text mode, upload local images, can be customized with css).

 If someone knows one please add a comment in this thread. I'm also adding pinning the issue so more people can see it.

**Comment by kenshin-takahiro** [comment]
> I'd appreciate any suggestions on editors that works similarly as the one we currently use (must have markdown and rich text mode, upload local images, can be customized with css).
> 
> If someone knows one please add a comment in this thread. I'm also adding pinning the issue so more people can see it.

The built-in text/md editor in Nextcloud is pretty full featured, although I don't think it has rich text, but it has everything else. I think it is this one here: [https://github.com/nextcloud/text](https://github.com/nextcloud/text)
It looks very good:
![image](https://github.com/BaldissaraMatheus/Tasks.md/assets/157710434/d2e48aa9-673c-4909-a573-f2cd0bf60003)

**Comment by benjaminapetersen** [comment]
Hopping on this issue! I just found Tasks.md today and am assessing it as I prefer to maintain most of my tasks and notes manually.  However, the web UI this renders out is really handy. 

The editor escaping`- [ ]` into `- \[ \]` is unfortunate and a blocker for me currently, but will be watching this as it sounds like the feature is high on the implementation list!

**Comment by BaldissaraMatheus** [comment]
Shipped in the latest docker V3 release.
I'll close this issue now, if you encounter any problem feel free to open it again

**Comment by benjaminapetersen** [comment]
Commented on another issue, I see `bullet` and `number` lists when updating to `3.0.9`, but not `checkboxes`.  Manually creating items seems to still `- \[ \]` escape them on saves.  Curious if there is perhaps a config knob?
